pkg/operator/resource/resourceapply/core.go

```diff
@@ -210,14 +214,29 @@ func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, re
 	existingCopy := existing.DeepCopy()
 
 	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, required.ObjectMeta)
+
 	dataSame := equality.Semantic.DeepEqual(existingCopy.Data, required.Data)
 	if dataSame && !*modified {
 		return existingCopy, false, nil
 	}
 	existingCopy.Data = required.Data
 
 	if klog.V(4) {
-		klog.Infof("Secret %q changes: %v", required.Namespace+"/"+required.Name, JSONPatch(existing, required))
+		safeRequired := required.DeepCopy()
+		safeExisting := existing.DeepCopy()
+
+		for s := range safeExisting.Data {
+			safeExisting.Data[s] = []byte("OLD")
```

Comment:
maybe we want `OLD`, `NEW` and `MODIFIED` as the states to track?

Reply:
what do you mean with 3 states? We only have two.

Reply:
Ic what you mean. But note that we create a patch below. We cannot have OLD and NEW for all fields, only those which changed.